web-common/src/features/dashboards/time-dimension-details/charts/utils.ts

```diff
@@ -22,37 +29,57 @@ export function getVegaSpecForTDD(
   chartType: TDDAlternateCharts,
   expandedMeasureName: string,
   measureLabel: string,
+  isTimeComparison: boolean,
   isDimensional: boolean,
   dimensionName: string | undefined,
   comparedValues: (string | null)[] | undefined,
 ): VisualizationSpec {
-  const temporalFields = [{ name: "ts", label: "Time" }];
-  const measureFields = [{ name: expandedMeasureName, label: measureLabel }];
-  const nominalFields = [
-    {
-      name: "dimension",
-      label: dimensionName || "dimension",
-      values: comparedValues,
-    },
+  const temporalFields: ChartField[] = [{ name: "ts", label: "Time" }];
+  const measureFields: ChartField[] = [
+    { name: expandedMeasureName, label: measureLabel },
   ];
 
+  let nominalFields: ChartField[] = [];
+  if (isDimensional) {
+    nominalFields = [
+      {
+        name: "dimension",
+        label: dimensionName || "dimension",
+        values: comparedValues,
+      },
+    ];
+  } else if (isTimeComparison) {
+    nominalFields = [
+      {
+        name: "key",
+        label: "Comparing",
+        values: [expandedMeasureName, `comparison\\.${expandedMeasureName}`],
+      },
+    ];
+    temporalFields[0].tooltipName = "time";
+    measureFields[0].name = "measure";
```

Comment:
Similarly, I'm not following why we override the measure and dimension names `measure` and `dimension` (line 46). It looks like a requirement of the VegaLite spec deeper in the function tree?

Reply:
`measure` is used because for time comparisons the accessor is different for current period and comparison period.
`dimension` is used to avoid special characters which might be present in the dimension name.